Doc/c-api/type.rst

```diff
@@ -57,6 +57,48 @@ Type Objects
    modification of the attributes or base classes of the type.
 
 
+.. c:function:: int PyType_AddWatcher(PyType_WatchCallback callback)
```

Comment:
is there some place in the docs for more free-form prose about the type-watching mechanism?

maybe even a dedicated docs section for "watchers" in general (type, dict, the other ones if they are approved as well).

other than conceptual and high-level documentation, I feel like that would be a good place to say something like "well behaved extensions must never watch or clear watchers using watcher IDs that were not returned to them by the corresponding watcher registration API" (which you could also add here)

Reply:
+1 for a watcher how-to.

Reply:
I'm open to writing up a watcher howto if people feel that would be useful despite the limited audience for watchers, but I would prefer to reduce churn in its text by doing it as a separate PR after the various kinds of watchers are all merged, so it can discuss the overall picture without needing to be adjusted in each PR.

Added notes in the reference docs for `PyType_Watch` and `PyType_ClearWatcher` about not passing in a watcher ID you don't own.